c/models/utils.h

```diff
@@ -21,12 +21,87 @@
 //------------------------------------------------------------------------------
 #ifndef dt_MODELS_UTILS_h
 #define dt_MODELS_UTILS_h
+#include "py_datatable.h"
 
+template <typename T>
+using tptr = typename std::unique_ptr<T[]>;
+using uint64ptr = std::unique_ptr<uint64_t[]>;
 
+
+/*
+*  Helper template structures to convert C++ float/double types to
+*  datatable STypes::FLOAT32/STypes::FLOAT64. respectively.
+*/
+template<typename T> struct stype {
+  static void get_stype() {
+    throw TypeError() << "Only float and double types are supported";
+  }
+};
+
+
+template<> struct stype<float> {
+  static SType get_stype() {
+    return SType::FLOAT32;
+  }
+};
+
+
+template<> struct stype<double> {
+  static SType get_stype() {
+    return SType::FLOAT64;
+  }
+};
```

Comment:
Generally, there is no 1-to-1 correspondence between stypes and element types `T`.
We have template `element_t<stype>` (declared in column.h) that provides a mapping `stype -> T`, but the inverse is ill-defined, especially once we start adding new types such as datetime, uint64, categorical, array, etc.